{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# MNIST distributed training with tensorflow"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Set up the environment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import sagemaker\n",
    "from sagemaker import get_execution_role\n",
    "\n",
    "sagemaker_session = sagemaker.Session()\n",
    "\n",
    "role = get_execution_role()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Download the MNIST dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/ec2-user/anaconda3/envs/tensorflow_p27/lib/python2.7/site-packages/h5py/__init__.py:36: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.\n",
      "  from ._conv import register_converters as _register_converters\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Successfully downloaded train-images-idx3-ubyte.gz 9912422 bytes.\n",
      "Extracting data/train-images-idx3-ubyte.gz\n",
      "Successfully downloaded train-labels-idx1-ubyte.gz 28881 bytes.\n",
      "Extracting data/train-labels-idx1-ubyte.gz\n",
      "Successfully downloaded t10k-images-idx3-ubyte.gz 1648877 bytes.\n",
      "Extracting data/t10k-images-idx3-ubyte.gz\n",
      "Successfully downloaded t10k-labels-idx1-ubyte.gz 4542 bytes.\n",
      "Extracting data/t10k-labels-idx1-ubyte.gz\n",
      "('Writing', 'data/train.tfrecords')\n",
      "('Writing', 'data/validation.tfrecords')\n",
      "('Writing', 'data/test.tfrecords')\n"
     ]
    }
   ],
   "source": [
    "import utils\n",
    "from tensorflow.contrib.learn.python.learn.datasets import mnist\n",
    "import tensorflow as tf\n",
    "\n",
    "data_sets = mnist.read_data_sets('data', dtype=tf.uint8, reshape=False, validation_size=5000)\n",
    "\n",
    "utils.convert_to(data_sets.train, 'train', 'data')\n",
    "utils.convert_to(data_sets.validation, 'validation', 'data')\n",
    "utils.convert_to(data_sets.test, 'test', 'data')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Upload the data\n",
    "We use the ```sagemaker.Session.upload_data``` function to upload our datasets to an S3 location. The return value inputs identifies the location -- we will use this later when we start the training job."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:sagemaker:Created S3 bucket: sagemaker-us-west-2-766924284651\n"
     ]
    }
   ],
   "source": [
    "inputs = sagemaker_session.upload_data(path='data', key_prefix='data/DEMO-mnist')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Construct a script for distributed training "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "import os\r\n",
      "import tensorflow as tf\r\n",
      "from tensorflow.python.estimator.model_fn import ModeKeys as Modes\r\n",
      "\r\n",
      "INPUT_TENSOR_NAME = 'inputs'\r\n",
      "SIGNATURE_NAME = 'predictions'\r\n",
      "\r\n",
      "LEARNING_RATE = 0.001\r\n",
      "\r\n",
      "\r\n",
      "def model_fn(features, labels, mode, params):\r\n",
      "    # Input Layer\r\n",
      "    input_layer = tf.reshape(features[INPUT_TENSOR_NAME], [-1, 28, 28, 1])\r\n",
      "\r\n",
      "    # Convolutional Layer #1\r\n",
      "    conv1 = tf.layers.conv2d(\r\n",
      "        inputs=input_layer,\r\n",
      "        filters=32,\r\n",
      "        kernel_size=[5, 5],\r\n",
      "        padding='same',\r\n",
      "        activation=tf.nn.relu)\r\n",
      "\r\n",
      "    # Pooling Layer #1\r\n",
      "    pool1 = tf.layers.max_pooling2d(inputs=conv1, pool_size=[2, 2], strides=2)\r\n",
      "\r\n",
      "    # Convolutional Layer #2 and Pooling Layer #2\r\n",
      "    conv2 = tf.layers.conv2d(\r\n",
      "        inputs=pool1,\r\n",
      "        filters=64,\r\n",
      "        kernel_size=[5, 5],\r\n",
      "        padding='same',\r\n",
      "        activation=tf.nn.relu)\r\n",
      "    pool2 = tf.layers.max_pooling2d(inputs=conv2, pool_size=[2, 2], strides=2)\r\n",
      "\r\n",
      "    # Dense Layer\r\n",
      "    pool2_flat = tf.reshape(pool2, [-1, 7 * 7 * 64])\r\n",
      "    dense = tf.layers.dense(inputs=pool2_flat, units=1024, activation=tf.nn.relu)\r\n",
      "    dropout = tf.layers.dropout(\r\n",
      "        inputs=dense, rate=0.4, training=(mode == Modes.TRAIN))\r\n",
      "\r\n",
      "    # Logits Layer\r\n",
      "    logits = tf.layers.dense(inputs=dropout, units=10)\r\n",
      "\r\n",
      "    # Define operations\r\n",
      "    if mode in (Modes.PREDICT, Modes.EVAL):\r\n",
      "        predicted_indices = tf.argmax(input=logits, axis=1)\r\n",
      "        probabilities = tf.nn.softmax(logits, name='softmax_tensor')\r\n",
      "\r\n",
      "    if mode in (Modes.TRAIN, Modes.EVAL):\r\n",
      "        global_step = tf.train.get_or_create_global_step()\r\n",
      "        label_indices = tf.cast(labels, tf.int32)\r\n",
      "        loss = tf.losses.softmax_cross_entropy(\r\n",
      "            onehot_labels=tf.one_hot(label_indices, depth=10), logits=logits)\r\n",
      "        tf.summary.scalar('OptimizeLoss', loss)\r\n",
      "\r\n",
      "    if mode == Modes.PREDICT:\r\n",
      "        predictions = {\r\n",
      "            'classes': predicted_indices,\r\n",
      "            'probabilities': probabilities\r\n",
      "        }\r\n",
      "        export_outputs = {\r\n",
      "            SIGNATURE_NAME: tf.estimator.export.PredictOutput(predictions)\r\n",
      "        }\r\n",
      "        return tf.estimator.EstimatorSpec(\r\n",
      "            mode, predictions=predictions, export_outputs=export_outputs)\r\n",
      "\r\n",
      "    if mode == Modes.TRAIN:\r\n",
      "        optimizer = tf.train.AdamOptimizer(learning_rate=0.001)\r\n",
      "        train_op = optimizer.minimize(loss, global_step=global_step)\r\n",
      "        return tf.estimator.EstimatorSpec(mode, loss=loss, train_op=train_op)\r\n",
      "\r\n",
      "    if mode == Modes.EVAL:\r\n",
      "        eval_metric_ops = {\r\n",
      "            'accuracy': tf.metrics.accuracy(label_indices, predicted_indices)\r\n",
      "        }\r\n",
      "        return tf.estimator.EstimatorSpec(\r\n",
      "            mode, loss=loss, eval_metric_ops=eval_metric_ops)\r\n",
      "\r\n",
      "\r\n",
      "def serving_input_fn(params):\r\n",
      "    inputs = {INPUT_TENSOR_NAME: tf.placeholder(tf.float32, [None, 784])}\r\n",
      "    return tf.estimator.export.ServingInputReceiver(inputs, inputs)\r\n",
      "\r\n",
      "\r\n",
      "def read_and_decode(filename_queue):\r\n",
      "    reader = tf.TFRecordReader()\r\n",
      "    _, serialized_example = reader.read(filename_queue)\r\n",
      "\r\n",
      "    features = tf.parse_single_example(\r\n",
      "        serialized_example,\r\n",
      "        features={\r\n",
      "            'image_raw': tf.FixedLenFeature([], tf.string),\r\n",
      "            'label': tf.FixedLenFeature([], tf.int64),\r\n",
      "        })\r\n",
      "\r\n",
      "    image = tf.decode_raw(features['image_raw'], tf.uint8)\r\n",
      "    image.set_shape([784])\r\n",
      "    image = tf.cast(image, tf.float32) * (1. / 255)\r\n",
      "    label = tf.cast(features['label'], tf.int32)\r\n",
      "\r\n",
      "    return image, label\r\n",
      "\r\n",
      "\r\n",
      "def train_input_fn(training_dir, params):\r\n",
      "    return _input_fn(training_dir, 'train.tfrecords', batch_size=100)\r\n",
      "\r\n",
      "\r\n",
      "def eval_input_fn(training_dir, params):\r\n",
      "    return _input_fn(training_dir, 'test.tfrecords', batch_size=100)\r\n",
      "\r\n",
      "\r\n",
      "def _input_fn(training_dir, training_filename, batch_size=100):\r\n",
      "    test_file = os.path.join(training_dir, training_filename)\r\n",
      "    filename_queue = tf.train.string_input_producer([test_file])\r\n",
      "\r\n",
      "    image, label = read_and_decode(filename_queue)\r\n",
      "    images, labels = tf.train.batch(\r\n",
      "        [image, label], batch_size=batch_size,\r\n",
      "        capacity=1000 + 3 * batch_size)\r\n",
      "\r\n",
      "    return {INPUT_TENSOR_NAME: images}, labels\r\n"
     ]
    }
   ],
   "source": [
    "!cat 'mnist.py'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Create a training job using the sagemaker.TensorFlow estimator"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:sagemaker:Creating training-job with name: sagemaker-tensorflow-2018-04-15-02-09-49-209\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "........................................................\n",
      "\u001b[32m2018-04-15 02:14:27,077 INFO - root - running container entrypoint\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:27,078 INFO - root - starting train task\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:27,082 INFO - container_support.training - Training starting\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:27,756 INFO - root - running container entrypoint\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:27,756 INFO - root - starting train task\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:27,761 INFO - container_support.training - Training starting\u001b[0m\n",
      "\u001b[32m/usr/local/lib/python2.7/dist-packages/h5py/__init__.py:36: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.\n",
      "  from ._conv import register_converters as _register_converters\u001b[0m\n",
      "\u001b[31m/usr/local/lib/python2.7/dist-packages/h5py/__init__.py:36: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.\n",
      "  from ._conv import register_converters as _register_converters\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:28,978 INFO - botocore.vendored.requests.packages.urllib3.connectionpool - Starting new HTTP connection (1): 169.254.170.2\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:29,745 INFO - botocore.vendored.requests.packages.urllib3.connectionpool - Starting new HTTP connection (1): 169.254.170.2\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:29,961 INFO - botocore.vendored.requests.packages.urllib3.connectionpool - Starting new HTTPS connection (1): sagemaker-us-west-2-766924284651.s3.amazonaws.com\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:29,209 INFO - botocore.vendored.requests.packages.urllib3.connectionpool - Starting new HTTPS connection (1): sagemaker-us-west-2-766924284651.s3.amazonaws.com\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:29,559 INFO - botocore.vendored.requests.packages.urllib3.connectionpool - Starting new HTTPS connection (1): sagemaker-us-west-2-766924284651.s3-us-west-2.amazonaws.com\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:29,678 INFO - botocore.vendored.requests.packages.urllib3.connectionpool - Starting new HTTPS connection (2): sagemaker-us-west-2-766924284651.s3-us-west-2.amazonaws.com\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:29,804 INFO - botocore.vendored.requests.packages.urllib3.connectionpool - Starting new HTTPS connection (1): sagemaker-us-west-2-766924284651.s3.us-west-2.amazonaws.com\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:29,875 INFO - botocore.vendored.requests.packages.urllib3.connectionpool - Starting new HTTPS connection (2): sagemaker-us-west-2-766924284651.s3.us-west-2.amazonaws.com\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:30,038 INFO - botocore.vendored.requests.packages.urllib3.connectionpool - Starting new HTTPS connection (1): s3.amazonaws.com\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:30,343 INFO - botocore.vendored.requests.packages.urllib3.connectionpool - Starting new HTTPS connection (1): sagemaker-us-west-2-766924284651.s3-us-west-2.amazonaws.com\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:30,448 INFO - botocore.vendored.requests.packages.urllib3.connectionpool - Starting new HTTPS connection (2): sagemaker-us-west-2-766924284651.s3-us-west-2.amazonaws.com\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:30,465 INFO - botocore.vendored.requests.packages.urllib3.connectionpool - Starting new HTTPS connection (1): sagemaker-us-west-2-766924284651.s3.us-west-2.amazonaws.com\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:30,507 INFO - botocore.vendored.requests.packages.urllib3.connectionpool - Starting new HTTPS connection (2): sagemaker-us-west-2-766924284651.s3.us-west-2.amazonaws.com\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:30,637 INFO - botocore.vendored.requests.packages.urllib3.connectionpool - Starting new HTTPS connection (1): s3.amazonaws.com\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31,018 INFO - tf_container - ----------------------TF_CONFIG--------------------------\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31,018 INFO - tf_container - {\"environment\": \"cloud\", \"cluster\": {\"worker\": [\"algo-2:2222\"], \"ps\": [\"algo-1:2223\", \"algo-2:2223\"], \"master\": [\"algo-1:2222\"]}, \"task\": {\"index\": 0, \"type\": \"master\"}}\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31,018 INFO - tf_container - ---------------------------------------------------------\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31,018 INFO - tf_container - creating RunConfig:\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31,018 INFO - tf_container - {'save_checkpoints_secs': 300}\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31,018 INFO - tensorflow - TF_CONFIG environment variable: {u'environment': u'cloud', u'cluster': {u'ps': [u'algo-1:2223', u'algo-2:2223'], u'worker': [u'algo-2:2222'], u'master': [u'algo-1:2222']}, u'task': {u'index': 0, u'type': u'master'}}\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31,019 INFO - tf_container - creating the estimator\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31,020 INFO - tensorflow - Using config: {'_save_checkpoints_secs': 300, '_session_config': None, '_keep_checkpoint_max': 5, '_tf_random_seed': None, '_task_type': u'master', '_global_id_in_cluster': 0, '_is_chief': True, '_cluster_spec': <tensorflow.python.training.server_lib.ClusterSpec object at 0x7f843368af50>, '_model_dir': u's3://sagemaker-us-west-2-766924284651/sagemaker-tensorflow-2018-04-15-02-09-49-209/checkpoints', '_num_worker_replicas': 2, '_task_id': 0, '_log_step_count_steps': 100, '_master': u'grpc://algo-1:2222', '_save_checkpoints_steps': None, '_keep_checkpoint_every_n_hours': 10000, '_evaluation_master': '', '_service': None, '_save_summary_steps': 100, '_num_ps_replicas': 2}\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31,021 INFO - tensorflow - Start Tensorflow server.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31.042045: I tensorflow/core/distributed_runtime/rpc/grpc_channel.cc:215] Initialize GrpcChannelCache for job master -> {0 -> algo-1:2222}\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31.042045: I tensorflow/core/distributed_runtime/rpc/grpc_channel.cc:215] Initialize GrpcChannelCache for job master -> {0 -> localhost:2222}\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31.042475: I tensorflow/core/distributed_runtime/rpc/grpc_channel.cc:215] Initialize GrpcChannelCache for job ps -> {0 -> localhost:2223, 1 -> algo-2:2223}\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31.042489: I tensorflow/core/distributed_runtime/rpc/grpc_channel.cc:215] Initialize GrpcChannelCache for job ps -> {0 -> algo-1:2223, 1 -> algo-2:2223}\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31.042483: I tensorflow/core/distributed_runtime/rpc/grpc_channel.cc:215] Initialize GrpcChannelCache for job worker -> {0 -> algo-2:2222}\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31.042502: I tensorflow/core/distributed_runtime/rpc/grpc_channel.cc:215] Initialize GrpcChannelCache for job worker -> {0 -> algo-2:2222}\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31.044019: I tensorflow/core/distributed_runtime/rpc/grpc_server_lib.cc:324] Started server with target: grpc://localhost:2222\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31.044065: I tensorflow/core/distributed_runtime/rpc/grpc_server_lib.cc:324] Started server with target: grpc://localhost:2223\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31.061492: I tensorflow/core/platform/s3/aws_logging.cc:54] Initializing config loader against fileName /root//.aws/config and using profilePrefix = 1\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31.061537: I tensorflow/core/platform/s3/aws_logging.cc:54] Initializing config loader against fileName /root//.aws/credentials and using profilePrefix = 0\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31.062079: I tensorflow/core/platform/s3/aws_logging.cc:54] Setting provider to read credentials from /root//.aws/credentials for credentials file and /root//.aws/config for the config file , for use with profile default\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31.063303: I tensorflow/core/platform/s3/aws_logging.cc:54] Creating HttpClient with max connections2 and scheme http\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31.063336: I tensorflow/core/platform/s3/aws_logging.cc:54] Initializing CurlHandleContainer with size 2\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31.063352: I tensorflow/core/platform/s3/aws_logging.cc:54] Creating TaskRole with default ECSCredentialsClient and refresh rate 900000\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31.065303: I tensorflow/core/platform/s3/aws_logging.cc:54] Unable to open config file /root//.aws/credentials for reading.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31.066215: I tensorflow/core/platform/s3/aws_logging.cc:54] Failed to reload configuration.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31.066245: I tensorflow/core/platform/s3/aws_logging.cc:54] Unable to open config file /root//.aws/config for reading.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31.066255: I tensorflow/core/platform/s3/aws_logging.cc:54] Failed to reload configuration.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31.066267: I tensorflow/core/platform/s3/aws_logging.cc:54] Credentials have expired or will expire, attempting to repull from ECS IAM Service.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31.072163: I tensorflow/core/platform/s3/aws_logging.cc:54] Pool grown by 2\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31.073151: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:30,387 INFO - tf_container - ----------------------TF_CONFIG--------------------------\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:30,387 INFO - tf_container - {\"environment\": \"cloud\", \"cluster\": {\"worker\": [\"algo-2:2222\"], \"ps\": [\"algo-1:2223\", \"algo-2:2223\"], \"master\": [\"algo-1:2222\"]}, \"task\": {\"index\": 0, \"type\": \"worker\"}}\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:30,387 INFO - tf_container - ---------------------------------------------------------\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:30,388 INFO - tf_container - creating RunConfig:\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:30,388 INFO - tf_container - {'save_checkpoints_secs': 300}\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:30,388 INFO - tensorflow - TF_CONFIG environment variable: {u'environment': u'cloud', u'cluster': {u'ps': [u'algo-1:2223', u'algo-2:2223'], u'worker': [u'algo-2:2222'], u'master': [u'algo-1:2222']}, u'task': {u'index': 0, u'type': u'worker'}}\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:30,389 INFO - tf_container - creating the estimator\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:30,389 INFO - tensorflow - Using config: {'_save_checkpoints_secs': 300, '_session_config': None, '_keep_checkpoint_max': 5, '_tf_random_seed': None, '_task_type': u'worker', '_global_id_in_cluster': 1, '_is_chief': False, '_cluster_spec': <tensorflow.python.training.server_lib.ClusterSpec object at 0x7f550120af50>, '_model_dir': u's3://sagemaker-us-west-2-766924284651/sagemaker-tensorflow-2018-04-15-02-09-49-209/checkpoints', '_num_worker_replicas': 2, '_task_id': 0, '_log_step_count_steps': 100, '_master': u'grpc://algo-2:2222', '_save_checkpoints_steps': None, '_keep_checkpoint_every_n_hours': 10000, '_evaluation_master': '', '_service': None, '_save_summary_steps': 100, '_num_ps_replicas': 2}\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:30,390 INFO - tensorflow - Start Tensorflow server.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:30.413002: I tensorflow/core/distributed_runtime/rpc/grpc_channel.cc:215] Initialize GrpcChannelCache for job master -> {0 -> algo-1:2222}\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:30.413001: I tensorflow/core/distributed_runtime/rpc/grpc_channel.cc:215] Initialize GrpcChannelCache for job master -> {0 -> algo-1:2222}\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:30.413359: I tensorflow/core/distributed_runtime/rpc/grpc_channel.cc:215] Initialize GrpcChannelCache for job ps -> {0 -> algo-1:2223, 1 -> algo-2:2223}\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:30.413375: I tensorflow/core/distributed_runtime/rpc/grpc_channel.cc:215] Initialize GrpcChannelCache for job ps -> {0 -> algo-1:2223, 1 -> localhost:2223}\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:30.413405: I tensorflow/core/distributed_runtime/rpc/grpc_channel.cc:215] Initialize GrpcChannelCache for job worker -> {0 -> algo-2:2222}\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:30.413391: I tensorflow/core/distributed_runtime/rpc/grpc_channel.cc:215] Initialize GrpcChannelCache for job worker -> {0 -> localhost:2222}\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:30.414911: I tensorflow/core/distributed_runtime/rpc/grpc_server_lib.cc:324] Started server with target: grpc://localhost:2222\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:30.414978: I tensorflow/core/distributed_runtime/rpc/grpc_server_lib.cc:324] Started server with target: grpc://localhost:2223\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:30,415 INFO - tensorflow - Waiting 5 secs before starting training.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31.088745: I tensorflow/core/platform/s3/aws_logging.cc:54] Initializing CurlHandleContainer with size 25\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31.099273: I tensorflow/core/platform/s3/aws_logging.cc:54] Pool grown by 2\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31.099311: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31.182719: E tensorflow/core/platform/s3/aws_logging.cc:60] No response body. Response code: 404\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31.182765: W tensorflow/core/platform/s3/aws_logging.cc:57] If the signature check failed. This could be because of a time skew. Attempting to adjust the signer.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31.183516: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31,233 INFO - tensorflow - Calling model_fn.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31,356 WARNING - tensorflow - From /usr/local/lib/python2.7/dist-packages/tensorflow/python/ops/losses/losses_impl.py:731: softmax_cross_entropy_with_logits (from tensorflow.python.ops.nn_ops) is deprecated and will be removed in a future version.\u001b[0m\n",
      "\u001b[31mInstructions for updating:\n",
      "\u001b[0m\n",
      "\u001b[31mFuture major versions of TensorFlow will allow gradients to flow\u001b[0m\n",
      "\u001b[31minto the labels input on backprop by default.\n",
      "\u001b[0m\n",
      "\u001b[31mSee tf.nn.softmax_cross_entropy_with_logits_v2.\n",
      "\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31,596 INFO - tensorflow - Done calling model_fn.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31,598 INFO - tensorflow - Create CheckpointSaverHook.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31.598843: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31.605237: E tensorflow/core/platform/s3/aws_logging.cc:60] No response body. Response code: 404\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31.605265: W tensorflow/core/platform/s3/aws_logging.cc:57] If the signature check failed. This could be because of a time skew. Attempting to adjust the signer.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31.605407: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31.615751: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31.620520: E tensorflow/core/platform/s3/aws_logging.cc:60] No response body. Response code: 404\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31.620550: W tensorflow/core/platform/s3/aws_logging.cc:57] If the signature check failed. This could be because of a time skew. Attempting to adjust the signer.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31.620684: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31.632561: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31.639688: E tensorflow/core/platform/s3/aws_logging.cc:60] No response body. Response code: 404\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31.639717: W tensorflow/core/platform/s3/aws_logging.cc:57] If the signature check failed. This could be because of a time skew. Attempting to adjust the signer.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31.639856: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31.651517: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31.660800: I tensorflow/core/platform/s3/aws_logging.cc:54] Deleting file: /tmp/s3_filesystem_XXXXXX20180415T0214311523758471650\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31.713027: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31.721758: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31.730471: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31.736232: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31.744245: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31.836947: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31.921914: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:31.928344: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:32,044 INFO - tensorflow - Graph was finalized.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:32.045278: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:32.050129: E tensorflow/core/platform/s3/aws_logging.cc:60] No response body. Response code: 404\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:32.050165: W tensorflow/core/platform/s3/aws_logging.cc:57] If the signature check failed. This could be because of a time skew. Attempting to adjust the signer.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:32.050327: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:32.082746: I tensorflow/core/distributed_runtime/master_session.cc:1024] Start master session 50ec3021c82f57cb with config: allow_soft_placement: true\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:32,260 INFO - tensorflow - Running local_init_op.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:32,263 INFO - tensorflow - Done running local_init_op.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:32.316687: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:32.324272: E tensorflow/core/platform/s3/aws_logging.cc:60] No response body. Response code: 404\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:32.324308: W tensorflow/core/platform/s3/aws_logging.cc:57] If the signature check failed. This could be because of a time skew. Attempting to adjust the signer.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:32.324484: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:32.644442: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:32.721791: I tensorflow/core/platform/s3/aws_logging.cc:54] Deleting file: /tmp/s3_filesystem_XXXXXX20180415T0214321523758472643\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:32.722810: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:32.734133: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:32.793440: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:33,505 INFO - tensorflow - Saving checkpoints for 1 into s3://sagemaker-us-west-2-766924284651/sagemaker-tensorflow-2018-04-15-02-09-49-209/checkpoints/model.ckpt.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:33.528581: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:33.539286: I tensorflow/core/platform/s3/aws_logging.cc:54] Deleting file: /tmp/s3_filesystem_XXXXXX20180415T0214331523758473528\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:33.543049: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:33.576494: I tensorflow/core/platform/s3/aws_logging.cc:54] Deleting file: /tmp/s3_filesystem_XXXXXX20180415T0214331523758473539\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:33.576772: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:33.586763: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:33.626467: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:33.633380: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:33.648863: I tensorflow/core/platform/s3/aws_logging.cc:54] Deleting file: /tmp/s3_filesystem_XXXXXX20180415T0214331523758473633\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:33.649107: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:33.658808: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:33.683815: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32m2018-04-15 02:14:33.526753: I tensorflow/core/platform/s3/aws_logging.cc:54] Initializing config loader against fileName /root//.aws/config and using profilePrefix = 1\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:33.526800: I tensorflow/core/platform/s3/aws_logging.cc:54] Initializing config loader against fileName /root//.aws/credentials and using profilePrefix = 0\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:33.526817: I tensorflow/core/platform/s3/aws_logging.cc:54] Setting provider to read credentials from /root//.aws/credentials for credentials file and /root//.aws/config for the config file , for use with profile default\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:33.526830: I tensorflow/core/platform/s3/aws_logging.cc:54] Creating HttpClient with max connections2 and scheme http\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:33.526849: I tensorflow/core/platform/s3/aws_logging.cc:54] Initializing CurlHandleContainer with size 2\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:33.526864: I tensorflow/core/platform/s3/aws_logging.cc:54] Creating TaskRole with default ECSCredentialsClient and refresh rate 900000\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:33.526898: I tensorflow/core/platform/s3/aws_logging.cc:54] Unable to open config file /root//.aws/credentials for reading.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:33.526910: I tensorflow/core/platform/s3/aws_logging.cc:54] Failed to reload configuration.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:33.526921: I tensorflow/core/platform/s3/aws_logging.cc:54] Unable to open config file /root//.aws/config for reading.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:33.526929: I tensorflow/core/platform/s3/aws_logging.cc:54] Failed to reload configuration.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:33.526940: I tensorflow/core/platform/s3/aws_logging.cc:54] Credentials have expired or will expire, attempting to repull from ECS IAM Service.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:33.527033: I tensorflow/core/platform/s3/aws_logging.cc:54] Pool grown by 2\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:33.527051: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:33.533031: I tensorflow/core/platform/s3/aws_logging.cc:54] Initializing CurlHandleContainer with size 25\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:33.537485: I tensorflow/core/platform/s3/aws_logging.cc:54] Pool grown by 2\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:33.537540: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:33.585248: I tensorflow/core/platform/s3/aws_logging.cc:54] Deleting file: /tmp/s3_filesystem_XXXXXX20180415T0214331523758473534\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:33.630461: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:34.547630: I tensorflow/core/platform/s3/aws_logging.cc:54] Deleting file: /tmp/s3_filesystem_XXXXXX20180415T0214331523758473585\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:34.554758: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:34.566043: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:35.420650: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:35.452577: E tensorflow/core/platform/s3/aws_logging.cc:60] No response body. Response code: 404\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:35.452611: W tensorflow/core/platform/s3/aws_logging.cc:57] If the signature check failed. This could be because of a time skew. Attempting to adjust the signer.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:35.452758: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:35,499 INFO - tensorflow - Calling model_fn.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:35,621 WARNING - tensorflow - From /usr/local/lib/python2.7/dist-packages/tensorflow/python/ops/losses/losses_impl.py:731: softmax_cross_entropy_with_logits (from tensorflow.python.ops.nn_ops) is deprecated and will be removed in a future version.\u001b[0m\n",
      "\u001b[32mInstructions for updating:\n",
      "\u001b[0m\n",
      "\u001b[32mFuture major versions of TensorFlow will allow gradients to flow\u001b[0m\n",
      "\u001b[32minto the labels input on backprop by default.\n",
      "\u001b[0m\n",
      "\u001b[32mSee tf.nn.softmax_cross_entropy_with_logits_v2.\n",
      "\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:35.647578: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:35.655956: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:35.669032: I tensorflow/core/platform/s3/aws_logging.cc:54] Deleting file: /tmp/s3_filesystem_XXXXXX20180415T0214351523758475655\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:35.669251: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:35.680717: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:35.706023: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:35.716285: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:35.723774: I tensorflow/core/platform/s3/aws_logging.cc:54] Deleting file: /tmp/s3_filesystem_XXXXXX20180415T0214351523758475716\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:35.723983: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:35.732492: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:35.742125: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:35.752713: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:35.764567: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:35.774012: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:35.780421: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:35.789241: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:35.798599: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:35.806142: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:35.813940: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:35.822831: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:35,875 INFO - tensorflow - Done calling model_fn.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:35,877 INFO - tensorflow - Create CheckpointSaverHook.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:35,970 INFO - tensorflow - Graph was finalized.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:35.993537: I tensorflow/core/distributed_runtime/master_session.cc:1024] Start master session 0496690a21cb9215 with config: allow_soft_placement: true\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:36,048 INFO - tensorflow - Running local_init_op.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:36,052 INFO - tensorflow - Done running local_init_op.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:36,903 INFO - tensorflow - loss = 2.1466267, step = 1\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:37.417360: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:37.432253: I tensorflow/core/platform/s3/aws_logging.cc:54] Deleting file: /tmp/s3_filesystem_XXXXXX20180415T0214371523758477414\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:37.432542: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:37.444291: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:37.468399: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:37.508550: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:37.516691: E tensorflow/core/platform/s3/aws_logging.cc:60] No response body. Response code: 404\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:37.516727: W tensorflow/core/platform/s3/aws_logging.cc:57] If the signature check failed. This could be because of a time skew. Attempting to adjust the signer.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:37.516876: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:37.537922: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:37.563648: I tensorflow/core/platform/s3/aws_logging.cc:54] Deleting file: /tmp/s3_filesystem_XXXXXX20180415T0214371523758477537\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:37.563939: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:37.575178: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:37.605039: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:37.619380: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:37.625925: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:37.635181: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:37.642146: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:37.652962: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:37.660800: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:37.677482: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:37.683870: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:37.693487: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:37.703319: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:37,733 INFO - tensorflow - Calling model_fn.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:37,832 INFO - tensorflow - Done calling model_fn.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:37,851 INFO - tensorflow - Starting evaluation at 2018-04-15-02:14:37\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:37,910 INFO - tensorflow - Graph was finalized.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:37,911 INFO - tensorflow - Restoring parameters from s3://sagemaker-us-west-2-766924284651/sagemaker-tensorflow-2018-04-15-02-09-49-209/checkpoints/model.ckpt-1\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:37.932663: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:37.948641: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:37.955294: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:37.966420: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:37.971911: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:37.980923: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:37.987353: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:37.996093: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:38.008635: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:38.018542: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:38.030224: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:38.042523: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:38.083652: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:37.235482: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:37.243047: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:37.255302: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:37.294466: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:37.305623: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:37.353095: I tensorflow/core/platform/s3/aws_logging.cc:54] Deleting file: /tmp/s3_filesystem_XXXXXX20180415T0214371523758477305\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:37.353682: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:37.362036: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:37.372328: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:37.386509: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:14:37.396175: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:38.231055: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:38,258 INFO - tensorflow - Running local_init_op.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:38,264 INFO - tensorflow - Done running local_init_op.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:39,300 INFO - tensorflow - Evaluation [10/100]\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:40,252 INFO - tensorflow - Evaluation [20/100]\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:41,222 INFO - tensorflow - Evaluation [30/100]\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:42,132 INFO - tensorflow - Evaluation [40/100]\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:43,189 INFO - tensorflow - Evaluation [50/100]\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:44,069 INFO - tensorflow - Evaluation [60/100]\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:45,066 INFO - tensorflow - Evaluation [70/100]\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:45,947 INFO - tensorflow - Evaluation [80/100]\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:46,993 INFO - tensorflow - Evaluation [90/100]\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:47,883 INFO - tensorflow - Evaluation [100/100]\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:47,912 INFO - tensorflow - Finished evaluation at 2018-04-15-02:14:47\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:47,912 INFO - tensorflow - Saving dict for global step 1: accuracy = 0.314, global_step = 1, loss = 2.2199128\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:47.913387: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[31m2018-04-15 02:14:47.964233: E tensorflow/core/platform/s3/aws_logging.cc:60] No response body. Response code: 404\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:47.964361: W tensorflow/core/platform/s3/aws_logging.cc:57] If the signature check failed. This could be because of a time skew. Attempting to adjust the signer.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:47.964576: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.039004: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.062412: E tensorflow/core/platform/s3/aws_logging.cc:60] No response body. Response code: 404\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.062761: W tensorflow/core/platform/s3/aws_logging.cc:57] If the signature check failed. This could be because of a time skew. Attempting to adjust the signer.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.063076: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.073720: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.078727: E tensorflow/core/platform/s3/aws_logging.cc:60] No response body. Response code: 404\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.078905: W tensorflow/core/platform/s3/aws_logging.cc:57] If the signature check failed. This could be because of a time skew. Attempting to adjust the signer.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.079273: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.090797: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.101520: I tensorflow/core/platform/s3/aws_logging.cc:54] Deleting file: /tmp/s3_filesystem_XXXXXX20180415T0214481523758488090\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.144145: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.156502: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.167972: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.176188: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.185766: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.219066: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.250286: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.258244: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.276357: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.320907: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.470419: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.479059: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48,493 INFO - tensorflow - Calling model_fn.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48,542 INFO - tensorflow - Done calling model_fn.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48,542 INFO - tensorflow - Signatures INCLUDED in export for Classify: None\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48,542 INFO - tensorflow - Signatures INCLUDED in export for Regress: None\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48,542 INFO - tensorflow - Signatures INCLUDED in export for Predict: ['serving_default', 'predictions']\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.543143: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.549658: E tensorflow/core/platform/s3/aws_logging.cc:60] No response body. Response code: 404\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.549689: W tensorflow/core/platform/s3/aws_logging.cc:57] If the signature check failed. This could be because of a time skew. Attempting to adjust the signer.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.549831: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48,570 INFO - tensorflow - Restoring parameters from s3://sagemaker-us-west-2-766924284651/sagemaker-tensorflow-2018-04-15-02-09-49-209/checkpoints/model.ckpt-1\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.581882: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.598307: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.607381: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.616281: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.624957: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.636120: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.642033: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.649892: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.659024: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.665915: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.674308: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.684877: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.727833: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.875772: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.896241: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.902135: E tensorflow/core/platform/s3/aws_logging.cc:60] No response body. Response code: 404\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.902173: W tensorflow/core/platform/s3/aws_logging.cc:57] If the signature check failed. This could be because of a time skew. Attempting to adjust the signer.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.902409: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.916862: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.921692: E tensorflow/core/platform/s3/aws_logging.cc:60] No response body. Response code: 404\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.921719: W tensorflow/core/platform/s3/aws_logging.cc:57] If the signature check failed. This could be because of a time skew. Attempting to adjust the signer.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.921851: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.931402: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.937197: E tensorflow/core/platform/s3/aws_logging.cc:60] No response body. Response code: 404\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.937229: W tensorflow/core/platform/s3/aws_logging.cc:57] If the signature check failed. This could be because of a time skew. Attempting to adjust the signer.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.937363: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.948090: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.955450: E tensorflow/core/platform/s3/aws_logging.cc:60] No response body. Response code: 404\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.955475: W tensorflow/core/platform/s3/aws_logging.cc:57] If the signature check failed. This could be because of a time skew. Attempting to adjust the signer.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.955609: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.966414: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.971381: E tensorflow/core/platform/s3/aws_logging.cc:60] No response body. Response code: 404\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.971452: W tensorflow/core/platform/s3/aws_logging.cc:57] If the signature check failed. This could be because of a time skew. Attempting to adjust the signer.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.971747: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.987522: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.997877: I tensorflow/core/platform/s3/aws_logging.cc:54] Deleting file: /tmp/s3_filesystem_XXXXXX20180415T0214481523758488987\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:48.998077: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:49.005595: I tensorflow/core/platform/s3/aws_logging.cc:54] Deleting file: /tmp/s3_filesystem_XXXXXX20180415T0214481523758488997\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:49.005790: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:49.014652: I tensorflow/core/platform/s3/aws_logging.cc:54] Deleting file: /tmp/s3_filesystem_XXXXXX20180415T0214491523758489005\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:49,014 INFO - tensorflow - Assets added to graph.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:49,014 INFO - tensorflow - No assets to write.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:49.015221: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:49.021615: E tensorflow/core/platform/s3/aws_logging.cc:60] No response body. Response code: 404\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:49.021677: W tensorflow/core/platform/s3/aws_logging.cc:57] If the signature check failed. This could be because of a time skew. Attempting to adjust the signer.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:49.021901: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:49.032879: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:49.037724: E tensorflow/core/platform/s3/aws_logging.cc:60] No response body. Response code: 404\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:49.037754: W tensorflow/core/platform/s3/aws_logging.cc:57] If the signature check failed. This could be because of a time skew. Attempting to adjust the signer.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:49.037892: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:49.048627: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:49.054134: E tensorflow/core/platform/s3/aws_logging.cc:60] No response body. Response code: 404\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:49.054170: W tensorflow/core/platform/s3/aws_logging.cc:57] If the signature check failed. This could be because of a time skew. Attempting to adjust the signer.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:49.054334: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:49.066177: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:49.076564: I tensorflow/core/platform/s3/aws_logging.cc:54] Deleting file: /tmp/s3_filesystem_XXXXXX20180415T0214491523758489066\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:49.162580: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:49.172584: I tensorflow/core/platform/s3/aws_logging.cc:54] Deleting file: /tmp/s3_filesystem_XXXXXX20180415T0214491523758489162\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:49.185512: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:49.450540: I tensorflow/core/platform/s3/aws_logging.cc:54] Deleting file: /tmp/s3_filesystem_XXXXXX20180415T0214491523758489172\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:49.452548: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:49.464870: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:50.064597: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:50.075955: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:50.111011: I tensorflow/core/platform/s3/aws_logging.cc:54] Deleting file: /tmp/s3_filesystem_XXXXXX20180415T0214501523758490075\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:50.111267: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:50.121885: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:50.187582: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:50.198382: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:50.208406: I tensorflow/core/platform/s3/aws_logging.cc:54] Deleting file: /tmp/s3_filesystem_XXXXXX20180415T0214501523758490198\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:50.208583: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:50.219608: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:50.229646: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:50.240958: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:50.249847: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:50.260808: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:50.274064: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:50.452286: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:50.461799: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:50.569075: I tensorflow/core/platform/s3/aws_logging.cc:54] Deleting file: /tmp/s3_filesystem_XXXXXX20180415T0214501523758490461\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:50.569315: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:50.579614: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:50.591936: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:50.615692: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:50.620530: E tensorflow/core/platform/s3/aws_logging.cc:60] No response body. Response code: 404\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:50.620557: W tensorflow/core/platform/s3/aws_logging.cc:57] If the signature check failed. This could be because of a time skew. Attempting to adjust the signer.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:50.620686: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:50.634857: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:50.660404: I tensorflow/core/platform/s3/aws_logging.cc:54] Deleting file: /tmp/s3_filesystem_XXXXXX20180415T0214501523758490634\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:50,660 INFO - tensorflow - SavedModel written to: s3://sagemaker-us-west-2-766924284651/sagemaker-tensorflow-2018-04-15-02-09-49-209/checkpoints/export/Servo/temp-1523758488/saved_model.pb\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:50.660908: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:50.685660: E tensorflow/core/platform/s3/aws_logging.cc:60] No response body. Response code: 404\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:50.685690: W tensorflow/core/platform/s3/aws_logging.cc:57] If the signature check failed. This could be because of a time skew. Attempting to adjust the signer.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:50.685855: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:50.693231: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:50.707426: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:50.724953: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:50.734517: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:50.784424: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:50.795115: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:50.812819: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:50.822680: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:51.396236: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:51.403569: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:51.427113: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:51.437082: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:51.445231: E tensorflow/core/platform/s3/aws_logging.cc:60] No response body. Response code: 404\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:51.445263: W tensorflow/core/platform/s3/aws_logging.cc:57] If the signature check failed. This could be because of a time skew. Attempting to adjust the signer.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:51.445409: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:51.458853: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:51.467323: E tensorflow/core/platform/s3/aws_logging.cc:60] No response body. Response code: 404\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:51.467354: W tensorflow/core/platform/s3/aws_logging.cc:57] If the signature check failed. This could be because of a time skew. Attempting to adjust the signer.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:51.467493: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:51.477870: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:14:51,585 INFO - tensorflow - loss = 2.309563, step = 0\u001b[0m\n",
      "\u001b[31m2018-04-15 02:15:12,612 INFO - tensorflow - global_step/sec: 4.89814\u001b[0m\n",
      "\u001b[32m2018-04-15 02:15:16,282 INFO - tensorflow - loss = 0.06576669, step = 163 (39.379 sec)\u001b[0m\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[31m2018-04-15 02:15:29,832 INFO - tensorflow - loss = 0.051766116, step = 232 (38.247 sec)\u001b[0m\n",
      "\u001b[31m2018-04-15 02:15:32,779 INFO - tensorflow - global_step/sec: 5.00828\u001b[0m\n",
      "\u001b[31m2018-04-15 02:15:53,196 INFO - tensorflow - global_step/sec: 4.99576\u001b[0m\n",
      "\u001b[32m2018-04-15 02:15:58,742 INFO - tensorflow - loss = 0.03206717, step = 377 (42.460 sec)\u001b[0m\n",
      "\u001b[31m2018-04-15 02:16:07,545 INFO - tensorflow - loss = 0.054420207, step = 419 (37.712 sec)\u001b[0m\n",
      "\u001b[31m2018-04-15 02:16:13,893 INFO - tensorflow - global_step/sec: 4.92834\u001b[0m\n",
      "\u001b[31m2018-04-15 02:16:34,701 INFO - tensorflow - global_step/sec: 4.90195\u001b[0m\n",
      "\u001b[31m2018-04-15 02:16:34.702351: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:16:34.798755: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:16:34.870621: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:16:34.878916: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:16:41,165 INFO - tensorflow - loss = 0.03631803, step = 585 (42.423 sec)\u001b[0m\n",
      "\u001b[31m2018-04-15 02:16:46,778 INFO - tensorflow - loss = 0.043270744, step = 613 (39.234 sec)\u001b[0m\n",
      "\u001b[31m2018-04-15 02:16:55,122 INFO - tensorflow - global_step/sec: 4.94592\u001b[0m\n",
      "\u001b[31m2018-04-15 02:17:15,282 INFO - tensorflow - global_step/sec: 5.00992\u001b[0m\n",
      "\u001b[32m2018-04-15 02:17:22,390 INFO - tensorflow - loss = 0.032181706, step = 791 (41.225 sec)\u001b[0m\n",
      "\u001b[31m2018-04-15 02:17:25,754 INFO - tensorflow - loss = 0.057541665, step = 808 (38.975 sec)\u001b[0m\n",
      "\u001b[31m2018-04-15 02:17:35,960 INFO - tensorflow - global_step/sec: 4.98112\u001b[0m\n",
      "\u001b[31m2018-04-15 02:17:56,399 INFO - tensorflow - global_step/sec: 4.94146\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:04,506 INFO - tensorflow - loss = 0.014134713, step = 999 (38.753 sec)\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:04,511 INFO - tensorflow - Saving checkpoints for 1001 into s3://sagemaker-us-west-2-766924284651/sagemaker-tensorflow-2018-04-15-02-09-49-209/checkpoints/model.ckpt.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:04.514029: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:04.562467: I tensorflow/core/platform/s3/aws_logging.cc:54] Deleting file: /tmp/s3_filesystem_XXXXXX20180415T0218041523758684513\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:04.565625: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:04.584774: I tensorflow/core/platform/s3/aws_logging.cc:54] Deleting file: /tmp/s3_filesystem_XXXXXX20180415T0218041523758684562\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:04.585070: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:04.597796: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:04.629046: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:04.638873: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:04.678615: I tensorflow/core/platform/s3/aws_logging.cc:54] Deleting file: /tmp/s3_filesystem_XXXXXX20180415T0218041523758684638\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:04.679322: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:04.693090: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:04.720274: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:18:04.498195: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:18:04.585157: I tensorflow/core/platform/s3/aws_logging.cc:54] Deleting file: /tmp/s3_filesystem_XXXXXX20180415T0218041523758684497\u001b[0m\n",
      "\u001b[32m2018-04-15 02:18:04,605 INFO - tensorflow - loss = 0.01230221, step = 1000 (42.215 sec)\u001b[0m\n",
      "\u001b[32m2018-04-15 02:18:04,663 INFO - tensorflow - Loss for final step: 0.01230221.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:18:04.688287: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:18:05.517779: I tensorflow/core/platform/s3/aws_logging.cc:54] Deleting file: /tmp/s3_filesystem_XXXXXX20180415T0218041523758684585\u001b[0m\n",
      "\u001b[32m2018-04-15 02:18:05.523340: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:18:05.538263: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:18:06.060442: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:18:06.067600: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:18:06.099832: I tensorflow/core/platform/s3/aws_logging.cc:54] Deleting file: /tmp/s3_filesystem_XXXXXX20180415T0218061523758686067\u001b[0m\n",
      "\u001b[32m2018-04-15 02:18:06.100016: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:18:06.113933: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:18:06.134503: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:18:06.141052: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:18:06.149871: I tensorflow/core/platform/s3/aws_logging.cc:54] Deleting file: /tmp/s3_filesystem_XXXXXX20180415T0218061523758686140\u001b[0m\n",
      "\u001b[32m2018-04-15 02:18:06.150037: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:18:06.156228: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:18:06.167025: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:18:06.177754: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:18:06.185693: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:18:06.195925: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:18:06.202127: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:18:06.208900: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:18:06.220032: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:18:06.227183: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:18:06.240306: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:18:06.254966: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:07.671064: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:07.685347: I tensorflow/core/platform/s3/aws_logging.cc:54] Deleting file: /tmp/s3_filesystem_XXXXXX20180415T0218071523758687670\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:07.685576: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:07.697393: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:07.725409: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:07.763353: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:07.769323: E tensorflow/core/platform/s3/aws_logging.cc:60] No response body. Response code: 404\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:07.769359: W tensorflow/core/platform/s3/aws_logging.cc:57] If the signature check failed. This could be because of a time skew. Attempting to adjust the signer.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:07.769507: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:07.786616: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:07.890378: I tensorflow/core/platform/s3/aws_logging.cc:54] Deleting file: /tmp/s3_filesystem_XXXXXX20180415T0218071523758687786\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:07.890728: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:07.905306: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:07.938729: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:07,955 INFO - tensorflow - Skip the current checkpoint eval due to throttle secs (600 secs).\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:07.956386: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:07.987605: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:08.052229: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:08.059219: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:18:07.384732: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:18:07.391751: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:18:07.402060: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:18:07.459700: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:18:07.469061: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:18:07.607448: I tensorflow/core/platform/s3/aws_logging.cc:54] Deleting file: /tmp/s3_filesystem_XXXXXX20180415T0218071523758687468\u001b[0m\n",
      "\u001b[32m2018-04-15 02:18:07.607693: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:18:07.618523: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:18:07.626160: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:18:07.635771: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[32m2018-04-15 02:18:07.644853: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:08,086 INFO - tensorflow - Loss for final step: 0.014134713.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:08,087 INFO - tensorflow - Training has already ended. But the last eval is skipped due to eval throttle_secs. Now evaluating the final checkpoint.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:08.087613: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:08.095558: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:08.104932: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:08.111825: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:08.121766: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:08.131831: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:08.148090: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:08.156464: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:08.238252: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:08.247876: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:08,292 INFO - tensorflow - Calling model_fn.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:08,390 INFO - tensorflow - Done calling model_fn.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:08,408 INFO - tensorflow - Starting evaluation at 2018-04-15-02:18:08\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:08,466 INFO - tensorflow - Graph was finalized.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:08,466 INFO - tensorflow - Restoring parameters from s3://sagemaker-us-west-2-766924284651/sagemaker-tensorflow-2018-04-15-02-09-49-209/checkpoints/model.ckpt-1001\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:08.487390: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:08.508362: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:08.516203: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:08.527024: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:08.535387: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:08.639932: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:08.649430: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:08.658865: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:08.669328: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:08.680230: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:08.691158: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:08.706056: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:08.750782: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:08.919723: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:08,947 INFO - tensorflow - Running local_init_op.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:08,954 INFO - tensorflow - Done running local_init_op.\u001b[0m\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[31m2018-04-15 02:18:09,945 INFO - tensorflow - Evaluation [10/100]\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:10,895 INFO - tensorflow - Evaluation [20/100]\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:11,767 INFO - tensorflow - Evaluation [30/100]\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:12,814 INFO - tensorflow - Evaluation [40/100]\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:13,710 INFO - tensorflow - Evaluation [50/100]\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:14,562 INFO - tensorflow - Evaluation [60/100]\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:15,572 INFO - tensorflow - Evaluation [70/100]\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:16,506 INFO - tensorflow - Evaluation [80/100]\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:17,422 INFO - tensorflow - Evaluation [90/100]\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:18,369 INFO - tensorflow - Evaluation [100/100]\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:18,400 INFO - tensorflow - Finished evaluation at 2018-04-15-02:18:18\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:18,402 INFO - tensorflow - Saving dict for global step 1001: accuracy = 0.9861, global_step = 1001, loss = 0.045372747\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:18.403587: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:18.538312: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:18.594319: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:18.602569: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:18,616 INFO - tensorflow - Calling model_fn.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:18,665 INFO - tensorflow - Done calling model_fn.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:18,666 INFO - tensorflow - Signatures INCLUDED in export for Classify: None\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:18,666 INFO - tensorflow - Signatures INCLUDED in export for Regress: None\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:18,666 INFO - tensorflow - Signatures INCLUDED in export for Predict: ['serving_default', 'predictions']\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:18.666710: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:18.673797: E tensorflow/core/platform/s3/aws_logging.cc:60] No response body. Response code: 404\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:18.673825: W tensorflow/core/platform/s3/aws_logging.cc:57] If the signature check failed. This could be because of a time skew. Attempting to adjust the signer.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:18.673963: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:18,697 INFO - tensorflow - Restoring parameters from s3://sagemaker-us-west-2-766924284651/sagemaker-tensorflow-2018-04-15-02-09-49-209/checkpoints/model.ckpt-1001\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:18.709237: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:18.727983: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:18.735736: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:18.747196: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:18.754377: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:18.763629: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:18.770153: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:18.779808: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:18.789018: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:18.797198: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:18.806176: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:18.817062: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:18.854822: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:19.023287: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:19.041564: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:19.048243: E tensorflow/core/platform/s3/aws_logging.cc:60] No response body. Response code: 404\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:19.048285: W tensorflow/core/platform/s3/aws_logging.cc:57] If the signature check failed. This could be because of a time skew. Attempting to adjust the signer.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:19.048461: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:19.057504: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:19.062938: E tensorflow/core/platform/s3/aws_logging.cc:60] No response body. Response code: 404\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:19.062965: W tensorflow/core/platform/s3/aws_logging.cc:57] If the signature check failed. This could be because of a time skew. Attempting to adjust the signer.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:19.063091: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:19.072322: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:19.078534: E tensorflow/core/platform/s3/aws_logging.cc:60] No response body. Response code: 404\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:19.078561: W tensorflow/core/platform/s3/aws_logging.cc:57] If the signature check failed. This could be because of a time skew. Attempting to adjust the signer.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:19.078709: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:19.090296: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:19.098849: I tensorflow/core/platform/s3/aws_logging.cc:54] Deleting file: /tmp/s3_filesystem_XXXXXX20180415T0218191523758699090\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:19,099 INFO - tensorflow - Assets added to graph.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:19,099 INFO - tensorflow - No assets to write.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:19.099458: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:19.106602: E tensorflow/core/platform/s3/aws_logging.cc:60] No response body. Response code: 404\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:19.106633: W tensorflow/core/platform/s3/aws_logging.cc:57] If the signature check failed. This could be because of a time skew. Attempting to adjust the signer.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:19.106776: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:19.115887: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:19.121279: E tensorflow/core/platform/s3/aws_logging.cc:60] No response body. Response code: 404\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:19.121311: W tensorflow/core/platform/s3/aws_logging.cc:57] If the signature check failed. This could be because of a time skew. Attempting to adjust the signer.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:19.121465: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:19.129295: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:19.134563: E tensorflow/core/platform/s3/aws_logging.cc:60] No response body. Response code: 404\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:19.134589: W tensorflow/core/platform/s3/aws_logging.cc:57] If the signature check failed. This could be because of a time skew. Attempting to adjust the signer.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:19.134731: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:19.223225: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:19.234993: I tensorflow/core/platform/s3/aws_logging.cc:54] Deleting file: /tmp/s3_filesystem_XXXXXX20180415T0218191523758699223\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:19.255759: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:19.266662: I tensorflow/core/platform/s3/aws_logging.cc:54] Deleting file: /tmp/s3_filesystem_XXXXXX20180415T0218191523758699255\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:19.278000: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:19.534044: I tensorflow/core/platform/s3/aws_logging.cc:54] Deleting file: /tmp/s3_filesystem_XXXXXX20180415T0218191523758699266\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:19.536191: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:19.547604: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:19.790783: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:19.801629: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:19.816670: I tensorflow/core/platform/s3/aws_logging.cc:54] Deleting file: /tmp/s3_filesystem_XXXXXX20180415T0218191523758699801\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:19.816885: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:19.828749: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:19.854003: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:19.863956: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:19.874029: I tensorflow/core/platform/s3/aws_logging.cc:54] Deleting file: /tmp/s3_filesystem_XXXXXX20180415T0218191523758699863\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:19.874210: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:19.881335: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:19.890049: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:19.899122: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:19.908553: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:19.928307: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:19.939758: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:20.514283: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:20.531501: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:20.548245: I tensorflow/core/platform/s3/aws_logging.cc:54] Deleting file: /tmp/s3_filesystem_XXXXXX20180415T0218201523758700531\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:20.548436: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:20.556608: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:20.567447: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:20.591687: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:20.655444: E tensorflow/core/platform/s3/aws_logging.cc:60] No response body. Response code: 404\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:20.655486: W tensorflow/core/platform/s3/aws_logging.cc:57] If the signature check failed. This could be because of a time skew. Attempting to adjust the signer.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:20.655660: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:20.671433: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:20.708598: I tensorflow/core/platform/s3/aws_logging.cc:54] Deleting file: /tmp/s3_filesystem_XXXXXX20180415T0218201523758700671\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:20,708 INFO - tensorflow - SavedModel written to: s3://sagemaker-us-west-2-766924284651/sagemaker-tensorflow-2018-04-15-02-09-49-209/checkpoints/export/Servo/temp-1523758698/saved_model.pb\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:20.709157: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:20.715162: E tensorflow/core/platform/s3/aws_logging.cc:60] No response body. Response code: 404\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:20.715193: W tensorflow/core/platform/s3/aws_logging.cc:57] If the signature check failed. This could be because of a time skew. Attempting to adjust the signer.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:20.715340: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:20.724457: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:20.738971: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:20.752750: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:20.761389: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:20.786336: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:20.794873: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:20.812349: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:20.821354: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:21.422502: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:21.433566: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:21.466749: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:21.477978: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:21.484480: E tensorflow/core/platform/s3/aws_logging.cc:60] No response body. Response code: 404\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:21.484509: W tensorflow/core/platform/s3/aws_logging.cc:57] If the signature check failed. This could be because of a time skew. Attempting to adjust the signer.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:21.484654: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:21.497412: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:21.503897: E tensorflow/core/platform/s3/aws_logging.cc:60] No response body. Response code: 404\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:21.503924: W tensorflow/core/platform/s3/aws_logging.cc:57] If the signature check failed. This could be because of a time skew. Attempting to adjust the signer.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:21.504078: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:21.519114: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:21,532 INFO - botocore.vendored.requests.packages.urllib3.connectionpool - Starting new HTTPS connection (1): sagemaker-us-west-2-766924284651.s3.amazonaws.com\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:21,901 INFO - botocore.vendored.requests.packages.urllib3.connectionpool - Starting new HTTPS connection (1): sagemaker-us-west-2-766924284651.s3-us-west-2.amazonaws.com\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:21,940 INFO - botocore.vendored.requests.packages.urllib3.connectionpool - Starting new HTTPS connection (1): sagemaker-us-west-2-766924284651.s3.us-west-2.amazonaws.com\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:22,062 INFO - tf_container - Downloaded saved model at /opt/ml/model/export/Servo/1523758488/saved_model.pb\u001b[0m\n",
      "\u001b[31m2018-04-15 02:18:22,076 INFO - botocore.vendored.requests.packages.urllib3.connectionpool - Starting new HTTPS connection (2): sagemaker-us-west-2-766924284651.s3.us-west-2.amazonaws.com\u001b[0m\n",
      "\u001b[32m2018-04-15 02:18:24,861 INFO - tf_container - master algo-1 is down, stopping parameter server\u001b[0m\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "===== Job Complete =====\n",
      "Billable seconds: 733\n"
     ]
    }
   ],
   "source": [
    "from sagemaker.tensorflow import TensorFlow\n",
    "\n",
    "mnist_estimator = TensorFlow(entry_point='mnist.py',\n",
    "                             role=role,\n",
    "                             training_steps=1000, \n",
    "                             evaluation_steps=100,\n",
    "                             train_instance_count=2,\n",
    "                             train_instance_type='ml.c4.xlarge')\n",
    "\n",
    "mnist_estimator.fit(inputs)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The **```fit```** method will create a training job in two **ml.c4.xlarge** instances. The logs above will show the instances doing training, evaluation, and incrementing the number of **training steps**. \n",
    "\n",
    "In the end of the training, the training job will generate a saved model for TF serving."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "# Deploy the trained model to prepare for predictions\n",
    "\n",
    "The deploy() method creates an endpoint which serves prediction requests in real-time."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:sagemaker:Creating model with name: sagemaker-tensorflow-2018-04-15-02-09-49-209\n",
      "INFO:sagemaker:Creating endpoint with name sagemaker-tensorflow-2018-04-15-02-09-49-209\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------------------------!"
     ]
    }
   ],
   "source": [
    "mnist_predictor = mnist_estimator.deploy(initial_instance_count=1,\n",
    "                                             instance_type='ml.m4.xlarge')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Invoking the endpoint"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Successfully downloaded train-images-idx3-ubyte.gz 9912422 bytes.\n",
      "Extracting /tmp/data/train-images-idx3-ubyte.gz\n",
      "Successfully downloaded train-labels-idx1-ubyte.gz 28881 bytes.\n",
      "Extracting /tmp/data/train-labels-idx1-ubyte.gz\n",
      "Successfully downloaded t10k-images-idx3-ubyte.gz 1648877 bytes.\n",
      "Extracting /tmp/data/t10k-images-idx3-ubyte.gz\n",
      "Successfully downloaded t10k-labels-idx1-ubyte.gz 4542 bytes.\n",
      "Extracting /tmp/data/t10k-labels-idx1-ubyte.gz\n",
      "========================================\n",
      "label is 7\n",
      "prediction is 7\n",
      "========================================\n",
      "label is 2\n",
      "prediction is 2\n",
      "========================================\n",
      "label is 1\n",
      "prediction is 1\n",
      "========================================\n",
      "label is 0\n",
      "prediction is 0\n",
      "========================================\n",
      "label is 4\n",
      "prediction is 4\n",
      "========================================\n",
      "label is 1\n",
      "prediction is 1\n",
      "========================================\n",
      "label is 4\n",
      "prediction is 4\n",
      "========================================\n",
      "label is 9\n",
      "prediction is 9\n",
      "========================================\n",
      "label is 5\n",
      "prediction is 5\n",
      "========================================\n",
      "label is 9\n",
      "prediction is 9\n"
     ]
    }
   ],
   "source": [
    "import numpy as np\n",
    "from tensorflow.examples.tutorials.mnist import input_data\n",
    "\n",
    "mnist = input_data.read_data_sets(\"/tmp/data/\", one_hot=True)\n",
    "\n",
    "for i in range(10):\n",
    "    data = mnist.test.images[i].tolist()\n",
    "    tensor_proto = tf.make_tensor_proto(values=np.asarray(data), shape=[1, len(data)], dtype=tf.float32)\n",
    "    predict_response = mnist_predictor.predict(tensor_proto)\n",
    "    \n",
    "    print(\"========================================\")\n",
    "    label = np.argmax(mnist.test.labels[i])\n",
    "    print(\"label is {}\".format(label))\n",
    "    prediction = predict_response['outputs']['classes']['int64Val'][0]\n",
    "    print(\"prediction is {}\".format(prediction))"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "conda_tensorflow_p27",
   "language": "python",
   "name": "conda_tensorflow_p27"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.14"
  },
  "notice": "Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.  Licensed under the Apache License, Version 2.0 (the \"License\"). You may not use this file except in compliance with the License. A copy of the License is located at http://aws.amazon.com/apache2.0/ or in the \"license\" file accompanying this file. This file is distributed on an \"AS IS\" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License."
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
